### PR TITLE
firewalk: init at 5.0

### PR DESCRIPTION
**Pull request by Tochiaha**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/firewalk/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        hash = "sha256-d45Hm8WFTiVLPY2fPwnLdyapXLnBQaEJvzIgMdlzl08=";
+    };
+
+    buildInputs = [ libnet libpcap libdnet ];
+
+    meta = with lib; {
+        description = "A gateway ACL scanner";
+        homepage = "https://salsa.debian.org/pkg-security-team/firewalk";
+        licence = [  ];
+        maintainers = with maintainers; [  ];
```
Comment (afh) [inline pkgs/applications/networking/firewalk/default.nix:24]: I feel like leaving these out, unless they contain meaning values, is a better option here.

File: pkgs/applications/networking/firewalk/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        platforms = platforms.linux;
+
```
Comment (afh) [inline pkgs/applications/networking/firewalk/default.nix:26]: Remove unnecessary newline

```suggestion
```

File: pkgs/applications/networking/firewalk/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    src = fetchurl {
+        url = "https://salsa.debian.org/pkg-security-team/firewalk/-/archive/debian/master/firewalk-debian-master.tar.gz";
+        hash = "sha256-d45Hm8WFTiVLPY2fPwnLdyapXLnBQaEJvzIgMdlzl08=";
+    };
+
+    buildInputs = [ libnet libpcap libdnet ];
+
+    meta = with lib; {
+        description = "A gateway ACL scanner";
```
Comment (afh) [inline pkgs/applications/networking/firewalk/default.nix:21]: The indefinite article 'a' can be removed for brevity.

```suggestion
        description = "Gateway ACL scanner";
```

**Review by afh — COMMENTED** [review]
The commit message is very generic and would be much more useful if it said: "Add firewalk package".

Leaving a few other comments and suggestions below…

**Comment by jfvillablanca** [comment]
Can you also use the new by-name method of packaging
https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#readme
I think this is the prescribed way for new packages :)

**Comment by Tochiaha** [comment]
All necessary changes applied.

File: pkgs/applications/networking/firewalk/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+, stdenv
+, fetchurl
+, libnet
+, libpcap
+, libdnet
+}:
+
+stdenv.mkDerivation rec {
+    pname = "firewalk";
+    version = "5.0-5";
```
Comment (natsukium) [inline pkgs/applications/networking/firewalk/default.nix:11]:
```suggestion
    version = "5.0";
```
The number at the end is the debian revision number, so we don't need it.

**Comment by jfvillablanca** [comment]
> All necessary changes applied.

You also have to squash all of your commits (following contributions guideline) and git push force

You can check out the guideline to learn how to do it

File: pkgs/applications/networking/firewalk/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        url = "https://salsa.debian.org/pkg-security-team/firewalk/-/archive/debian/5.0-5/firewalk-debian-5.0-5.tar.gz";
+        hash = "sha256-luFTi4DnXbiNDI2uPgbtF4Y9m/SdxQBs+Dfdmck7mFU=";
+    };
+
+    buildInputs = [ libnet libpcap libdnet ];
+
+    meta = with lib; {
+        description = "Gateway ACL scanner";
+        homepage = "https://salsa.debian.org/pkg-security-team/firewalk";
+        platforms = platforms.linux;
```
Comment (natsukium) [inline pkgs/applications/networking/firewalk/default.nix:23]:
```suggestion
        platforms = platforms.linux;
        license = licenses.bsd2;
        maintainers = with maintainers; [ Tochiaha ];
```

File: pkgs/applications/networking/firewalk/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    src = fetchurl {
+        url = "https://salsa.debian.org/pkg-security-team/firewalk/-/archive/debian/5.0-5/firewalk-debian-5.0-5.tar.gz";
+        hash = "sha256-luFTi4DnXbiNDI2uPgbtF4Y9m/SdxQBs+Dfdmck7mFU=";
+    };
+
+    buildInputs = [ libnet libpcap libdnet ];
+
+    meta = with lib; {
+        description = "Gateway ACL scanner";
+        homepage = "https://salsa.debian.org/pkg-security-team/firewalk";
```
Comment (natsukium) [inline pkgs/applications/networking/firewalk/default.nix:22]:
```suggestion
        homepage = "http://packetfactory.openwall.net/projects/firewalk";
```
Please cite the upstream homepage.

**Review by natsukium — CHANGES_REQUESTED** [review]

**Pull request closed without merging**